components/account/CollectionModal.vue

```diff
@@ -0,0 +1,222 @@
+            {{ $t('collectionModal.createCollection') }}
+          </b-button>
+        </div>
+      </b-form>
+    </div>
+    <div v-else>
+      <b-button
+        variant="primary"
+        class="btn-collection w-100 mb-3 text-left"
+        @click="createCollection"
+      >
+        {{ $t('collectionModal.createNewCollection') }}
+      </b-button>
+      <div class="collections">
+        <b-button
+          v-for="(collection, index) in collections"
+          :key="index"
+          :style="buttonBackground(collection.thumbnail)"
+          variant="overlay"
+          class="btn-collection w-100 text-left"
+          @click="addItem(collection.id)"
+        >
+          <span>{{ displayField(collection, 'title') }} ({{ collection.visibility }}) - {{ $tc('items.itemCount', collection.total) }}</span>
+        </b-button>
+      </div>
+      <div class="modal-footer">
+        <b-button
+          variant="outline-primary"
+          @click="hideModal()"
+        >
+          {{ $t('actions.close') }}
+        </b-button>
+      </div>
+    </div>
+  </b-modal>
+</template>
+
+<script>
+  export default {
+    name: 'CollectionModal',
+
+    props: {
+      itemId: {
+        type: String,
+        required: true
+      },
+
+      modalId: {
+        type: String,
+        default: 'collection-modal'
+      }
+    },
+
+    data() {
+      return {
+        showForm: false,
+        newCollectionName: '',
+        newCollectionDescription: '',
+        newCollectionPrivate: false,
+        collections: []
+      };
+    },
+
+    mounted() {
+      this.$root.$on('bv::modal::hidden', () => {
+        this.showForm = false;
+      });
+    },
+
+    computed: {
+      title() {
+        return this.showForm ?
+          this.$t('collectionModal.createNewCollection') :
+          this.$t('collectionModal.addToCollection');
+      }
+    },
+
+    methods: {
+      async fetchCollections() {
+        const setIds = await this.$sets.getSetsByCreator(this.$auth.user.sub, '', 'minimal');
+        const setsNoImage = await this.$sets.getAllSets(setIds);
+        this.collections = await this.$sets.getSetImages(setsNoImage);
+      },
+      createCollection() {
+        this.showForm = true;
+      },
+      goBack() {
+        this.showForm = false;
+        this.newCollectionName = '';
+        this.newCollectionDescription = '';
+        this.newCollectionPrivate = false;
+      },
+      hideModal() {
+        this.$nextTick(() => {
+          this.$bvModal.hide(this.modalId);
+        });
+      },
+
+      submitForm() {
+        // TODO: Create set-plugin call for creating a new set
+        // TODO: Add this item (its id can be found in the store) to the new set
+
+        // TODO: should this emit an event and allow the parent component to toast?
+        this.$bvToast.show('new-collection-toast');
+        this.newCollectionName = '';
+        this.newCollectionDescription = '';
+        this.newCollectionPrivate = false;
```

Comment:
This seems a duplication, as the next call is goBack, which also sets these to empty or false.